page_analyzer/utils/db.py

```diff
@@ -85,22 +87,43 @@ def get_urls_data() -> List[Dict[str, Any]]:
     return actual_urls_data
 
 
-def generate_check_context() -> Dict[str, int | str]:
-    status_code = 200
-    h1 = ""
-    title = ""
-    description = ""
-    check_context: dict[str, int | str] = {
-        "response_code": status_code,
-        "h1": h1,
-        "title": title,
-        "description": description,
-    }
+def generate_check_context(url_id: int) -> Dict[str, int | str] | None:
+    conn = connect_db()
+    with conn.cursor() as cur:
+        cur.execute(
+            "SELECT name FROM urls WHERE id=%(int)s;",
+            {
+                "int": url_id,
+            },
+        )
+        url_name = cur.fetchone()[0]
+        status_code = get_url_status_code(str(url_name))
+        if status_code is None:
+            return None
+        if status_code != 200:
+            h1 = ""
+            title = ""
+            description = ""
+        else:
+            h1 = "some header"
+            title = "some title"
+            description = "some description"
+
+        check_context: dict[str, int | str] = {
+            "response_code": status_code,
+            "h1": h1,
+            "title": title,
+            "description": description,
+        }
+    conn.close()
     return check_context
 
 
 def create_check(url_id: int) -> None:
-    check_context = generate_check_context()
+    check_context = generate_check_context(url_id)
+    if check_context is None:
+        flash("Произошла ошибка при проверке", "danger")
+        return None
```

Comment:
I'd rather flash inside the endpoint code as a db layer is not necessarily connected to the API layer

wdyt about making this function return a boolean and flash in the endpoint code if this function returned False?